app/src/main/java/com/example/roommatefinder/RegistrationActivity.java

```diff
@@ -0,0 +1,116 @@
+        // if using medium website, go to authentication tab, about middle of page
+        firebaseAuth = FirebaseAuth.getInstance();
+
+        regButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (validate()) //if validate returns true
+                {               //then update data to database
+                    String user_email = userEmail.getText().toString().trim();
+                    String user_password = userPassword.getText().toString().trim();
+
+                    firebaseAuth.createUserWithEmailAndPassword(user_email, user_password).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                        @Override
+                        public void onComplete(@NonNull Task<AuthResult> task) {
+
+                            if (task.isComplete()){
+                                sendUserData();
+                                Toast.makeText(RegistrationActivity.this, "Registration Successful!", Toast.LENGTH_SHORT).show();
+                                startActivity(new Intent(RegistrationActivity.this, MainActivity.class));
+                            }
+                            else {
+                                Toast.makeText(RegistrationActivity.this, "Registration Failed!", Toast.LENGTH_SHORT).show();
+
+                            }
+                        }
+                    });
+                }
+            }
+        });
+
+        userLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                startActivity(new Intent(RegistrationActivity.this, MainActivity.class));
+            }
+        });
+    }
+
+    private void setupUIViews() {
+        userName = (EditText) findViewById(R.id.etUserName);
+        userEmail = (EditText) findViewById(R.id.etUserEmail);
+        userPassword = (EditText) findViewById(R.id.etUserPassword);
+        regButton = (Button) findViewById(R.id.btnRegister);
+        userLogin = (TextView) findViewById(R.id.tvUserLogin);
+        userGender = (EditText) findViewById(R.id.etUserGender);
+        userClass = (EditText) findViewById(R.id.etUserClass);
+        userProfilePic = (ImageView) findViewById(R.id.ivProfile);
+
+    }
+
+    private Boolean validate() {
+        Boolean result = false;
+
+        name = userName.getText().toString();
+        email = userEmail.getText().toString();
+        password = userPassword.getText().toString();
+        gender = userGender.getText().toString();
+        classYear = userClass.getText().toString();
+
+        if (name.isEmpty() || email.isEmpty() || password.isEmpty() || gender.isEmpty() || classYear.isEmpty()) {
+            Toast.makeText(this, "Please fill out all the fields", Toast.LENGTH_SHORT).show();
+        }
+        else {
+            result = true;
+        }
+        return result;
```

Comment:
Sam, this is nit picking but instead of having a variable result to return a boolean, just return the boolean value